### PR TITLE
feat(graph2): Abstract CategoryVertex from Vertex

### PR DESCRIPTION
**Pull request by jbrundage**
Signed-off-by: Jaman Brundage <jbrundage372@gmail.com>

## Checklist:
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit message includes a "fixes" reference if appropriate.
  - [x] The commit is signed.
- [x] The change has been fully tested:
  - [x] I have viewed all related gallery items
  - [x] I have viewed all related dermatology items
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised new issues to address them separately

### CODE REVIEW

File: packages/graph/src/graph2/graph.ts [hunk 1]
```diff
@@ -550,6 +550,15 @@ export class Graph2 extends SVGZoomWidget {
...
 
+    private _categoryRenderer: FunctionComponent<Vertex> = CategoryVertex;
+    categoryRenderer(): FunctionComponent;
+    categoryRenderer(_: FunctionComponent): this;
+    categoryRenderer(_?: FunctionComponent): this | FunctionComponent {
+        if (!arguments.length) return this._categoryRenderer;
+        this._categoryRenderer = _;
+        return this;
+    }
+
```
Comment (GordonSmith) [inline packages/graph/src/graph2/graph.ts]: Not needed - Graoh2 should not be changed

**Comment by GordonSmith** [comment]
@jbrundage bump

**Comment by GordonSmith** [comment]
@jbrundage bump

File: packages/graph/src/graph2/graph.ts (hunk 1 above)
Comment (GordonSmith) [reply packages/graph/src/graph2/graph.ts:561]: @jbrundage bump

**Comment by jbrundage** [comment]
@GordonSmith I'm trying to wrap my mind around making this change without modifying graph2. 

The way I imagine this problem being solved is by moving the decision between two rendering modes into graph2. 
`_vertexRenderer` when a node doesn't have a categoryID
`_categoryRenderer` when a node does have a categoryID

This would allow users to specify a renderer for each case. Thoughts?

**Comment by GordonSmith** [comment]
I don't think a single graph instance would mix and match between category vertices and plain vertices?

Also after some usage - the Category Vertex didn't add any performance enhancement and brought general pain (selection colors for example).

**Comment by jbrundage** [comment]
Interesting (@ the fact that the category vertex version didn't add performance enhancements). I'll revise this PR when I finish with the latest Bar enhancements.

**Pull request closed without merging**